jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -700,12 +759,120 @@ ecma_op_typedarray_from (ecma_value_t items_val, /**< the source array-like obje
   }
 
+  /* 6 */
+  ecma_value_t using_iterator = ecma_op_get_method_by_symbol_id (items_val, LIT_MAGIC_STRING_ITERATOR);
+
+  /* 7 */
+  if (ECMA_IS_VALUE_ERROR (using_iterator))
+  {
+    return using_iterator;
+  }
+
+  ecma_value_t ret_value = ECMA_VALUE_ERROR;
+
+  /* 8 */
+  if (!ecma_is_value_undefined (using_iterator))
+  {
+    /* 8.a */
+    ecma_value_t iterator = ecma_op_get_iterator (items_val, using_iterator);
+    ecma_free_value (using_iterator);
+
+    /* 8.b */
+    if (ECMA_IS_VALUE_ERROR (iterator))
+    {
+      return iterator;
+    }
+
+    /* 8.c */
+    ecma_collection_t *values_p = ecma_new_collection ();
+
+    /* 8.e */
+    while (true)
+    {
+      /* 8.e.i */
+      ecma_value_t next = ecma_op_iterator_step (iterator);
+
+      /* 8.e.ii */
+      if (ECMA_IS_VALUE_ERROR (next))
+      {
+        ecma_free_value (iterator);
+        ecma_collection_free (values_p);
+        return next;
+      }
+
+      if (next == ECMA_VALUE_FALSE)
+      {
+        break;
+      }
+
+      /* 8.e.iii */
+      ecma_value_t next_value = ecma_op_iterator_value (next);
+      ecma_free_value (next);
+
+      if (ECMA_IS_VALUE_ERROR (next_value))
+      {
+        ecma_free_value (iterator);
+        ecma_collection_free (values_p);
+        return next_value;
+      }
+
+      ecma_collection_push_back (values_p, next_value);
+    }
+
+    ecma_free_value (iterator);
+
+    /* 8.g */
+    ecma_value_t new_typedarray = ecma_typedarray_create_object_with_length (values_p->item_count,
+                                                                             NULL,
+                                                                             proto_p,
+                                                                             element_size_shift,
+                                                                             typedarray_id);
+
+    /* 8.h */
+    if (ECMA_IS_VALUE_ERROR (new_typedarray))
+    {
+      ecma_collection_free (values_p);
+      return new_typedarray;
+    }
+
+    ecma_object_t *new_typedarray_p = ecma_get_object_from_value (new_typedarray);
+    ecma_typedarray_info_t info = ecma_typedarray_get_info (new_typedarray_p);
+    ecma_typedarray_setter_fn_t setter_cb = ecma_get_typedarray_setter_fn (info.id);
+
+    /* 8.j */
+    for (uint32_t index = 0; index < values_p->item_count; index++)
+    {
+      ecma_value_t error_val = ecma_op_typedarray_from_helper (this_val,
+                                                               values_p->buffer_p[index],
+                                                               index,
+                                                               func_object_p,
+                                                               info,
+                                                               setter_cb);
+
+      if (ECMA_IS_VALUE_ERROR (error_val))
+      {
+        ecma_deref_object (new_typedarray_p);
+        ecma_collection_destroy (values_p);
```

Comment:
before the collection_destroy the remaining elements in the collection must be freed.